### DISCUSSION TITLE
Control blinds with 2 different button press types and inching?

### DISCUSSION THREAD

**Discussion by LachNed** [RavenSystem/esp-homekit-devices#1845] [MEPLHAA Scripts]
Does anybody have an idea how to control my blinds with a Sonoff T1 2CH with 2 different actions...
1: long press up or down for approx. 1 second the blinds move up or down totally
2. short press up or down button while moving the blinds will stop and move for 0.5 seconds in the opposite direction.
3. to rotate the slats from horizontal to diagonal to vertical orientation | / — or just a little bit (to let more or less light in) I want to short press the up or down button for 0.5 seconds one or a few times

I don't know how to use the "t" press type and the "I" for inching with 0.5 seconds onto the same button (up/down)

can anybody help me? Or give me an example for a JSON?

Any help is appreciated
Thanks a lot

**Comment by paulzhdanov** [comment]
There is no such incoming attribute as "time of press" in this firmware. But for instance Aqara has a remote button with different reactions: short press, long press, double click, shake... Homekit automations can be done with scripts (shortcuts) to do exactly you want. The only issue is how to tie it to your blinds controller
